scylla/src/frame/response/result.rs

```diff
@@ -521,11 +527,16 @@ fn deser_prepared_metadata(buf: &mut &[u8]) -> StdResult<PreparedMetadata, Parse
     let col_count = types::read_int_length(buf)? as usize;
 
     let pk_count: usize = types::read_int(buf)?.try_into()?;
+    assert!(pk_count <= u16::MAX as usize);
```

Comment:
This assertion will result in panicking based on data received from the network, so it's a DoS risk. I suggest either dropping it or downgrading it to a ParseError

Reply:
You're right, I've put this assertion because of the cast of `i` below, but it's indeed not necessary. So I will drop it.